### DISCUSSION TITLE
Filter json data with a schema

### DISCUSSION THREAD

**Discussion by Fasi27** [networknt/json-schema-validator#1112] [Q&A] [answered]
Is it possible to filter json data with json-schema-validator?
I have a json data like:
data.json
```
{
  "format": "A4",
  "year": 2001,
  "tags": [
    "foo",
    "bar"
  ],
  "title": "title",
  "flag": true
}
```
and with filter1 one i like to extract the fields: format and title.
filter1.json
```
{
  "$schema": "http://json-schema.org/draft-04/schema",
  "additionalProperties": false,
  "properties": {
    "format": {
      "enum": [
        "A2",
        "A3",
        "A4"
      ],
      "type": "string"
    },
    "title": {
      "type": "string"
    }
  },
  "type": "object"
}
```

and with filter2 i like to get tags, title and flag
filter2.json
```
{
  "$schema": "http://json-schema.org/draft-04/schema",
  "additionalProperties": false,
  "properties": {
    "tags": {
      "items": {
        "type": "string"
      },
      "type": "array"
    },
    "title": {
      "type": "string"
    },
    "flag": {
      "type": "boolean"
    }
  },
  "type": "object"
}
```

Is there an easy way to do this?

**Comment by justin-tay** [comment]
You can see if you can implement that using a walker see https://github.com/networknt/json-schema-validator/blob/master/doc/walkers.md.

**Comment by Fasi27** [comment, marked as answer]
[JsonSchemaFilterWalkTest.java.txt](https://github.com/user-attachments/files/16964079/JsonSchemaFilterWalkTest.java.txt)

not perfect but working walker filtering the data
